src/AcceptanceTests/Receiving/When_messages_contains_typed_properties.cs

```diff
@@ -0,0 +1,115 @@
+                        message.Properties["CustomIntHeader"] = 200;
+                        message.Properties["CustomBoolHeader"] = true;
+                        message.Properties["CustomStringHeader"] = "Custom";
+                        message.Properties["CustomEmptyStringHeader"] = "";
+                        // ReSharper disable once RedundantCast
+                        message.Properties["CustomNullStringHeader"] = (string)null;
+                        message.Properties["CustomNullHeader"] = null;
+                        await sender.SendAsync(message);
+                    }
+                    finally
+                    {
+                        await sender.CloseAsync();
+                    }
+                }))
+                .Done(ctx => ctx.Received)
+                .Run();
+
+            Assert.That(context.Headers, Does.Not.ContainKey(BrokeredMessageHeaders.EstimatedMessageSize));
+            Assert.That(context.Headers, Does.Not.ContainKey(BrokeredMessageHeaders.TransportEncoding));
+            Assert.That(context.Headers, Does.ContainKey("CustomIntHeader").And.ContainValue("200"));
+            Assert.That(context.Headers, Does.ContainKey("CustomBoolHeader").And.ContainValue("True"));
+            Assert.That(context.Headers, Does.ContainKey("CustomStringHeader").And.ContainValue("Custom"));
+            Assert.That(context.Headers, Does.ContainKey("CustomEmptyStringHeader").And.ContainValue(""));
+            Assert.That(context.Headers, Does.ContainKey("CustomNullStringHeader").And.ContainValue(null));
+            Assert.That(context.Headers, Does.ContainKey("CustomNullHeader").And.ContainValue(null));
+        }
+
+        MessagingFactory factory;
+
+        static readonly string connectionString = TestUtility.DefaultConnectionString;
+
+        class Context : ScenarioContext
+        {
+            public bool Received { get; set; }
+            public Dictionary<string, string> Headers { get; set; } = new Dictionary<string, string>();
+        }
+
+
+        class Receiver : EndpointConfigurationBuilder
+        {
+            public Receiver()
+            {
+                EndpointSetup<DefaultServer>(config => { });
+            }
+
+            class SatelliteFeature : Feature
+            {
+                public SatelliteFeature()
+                {
+                    EnableByDefault();
+                }
+
+                protected override void Setup(FeatureConfigurationContext context)
+                {
+                    context.AddSatelliteReceiver("customSatellite", $"{Conventions.EndpointNamingConvention(typeof(Receiver))}-satellite", new PushRuntimeSettings(1), DefaultRecoverabilityPolicy.Invoke, Handle);
+                }
+
+                Task Handle(IBuilder builder, MessageContext context)
+                {
+                    var scenarioContext = builder.Build<Context>();
```

Comment:
Should we add a guard that ignores messages with acceptance test id different from scenario context testrunid? Satellites don't use the pipeline and thus the behaviors we have would not apply here

Reply:
Yeah. It's a sattelite. I forgot that a sattelite is a different queue and messages with a different test run id won't be filtered out. Better have that to avoid "flakiness".👍

Reply:
Done